### ISSUE TITLE
Weird artifacts on maps with alot of water when using Clean Textures mod

### ISSUE THREAD

**Issue by Gfiti** [JonathanTroyer/Progress-Renderer#27]
Image: https://i.gyazo.com/7f565da432e92ea12383471f581a5741.jpg
Mod used: https://steamcommunity.com/sharedfiles/filedetails/?id=2865361569

Mod changes how the water is rendered, adding reflections that move across the map as the time of day passes. Looks okay on most parts of the rendered image.

Around my base however you see this weird artifact that appears super smooth. Note how it mimics the shape of my base, albeit a bit warped. Across my playthorugh this warping is vastly different across all renders, it always seems to mimic the shape of my base though.

I went through the mod options, rendering as png or jpg, turning all options on or off but it changed nothing.

**Comment by TheVillageGuy** [comment]
So these artifacts are visible during gameplay?

**Comment by Gfiti** [comment]
No. Not at all. Wouldn't be asking here if I would also see that during the gameplay.

**Comment by TheVillageGuy** [comment]
Ok, I wasnt sure because you said across my gameplay. @JonathanTroyer is there a way to fix this?

**Comment by JonathanTroyer** [comment]
Hmmm, it's not immediately obvious what's going on. I'd have to see how the mod is performing rendering, it looks like the capture happened in the middle of a water rendering cycle. If that's the case, it'd be up to the mod authors to check if a map rendering is happening and not start a new water render if so.

**Comment by TheVillageGuy** [comment]
I am unable to look at it, but my best guess is the textures used are larger then one tile, so large in fact that they f can't fit on a screen, which prevents this from happening in hand, but not large enough to fill a render (they'd technically have to be infinite). So what we're seeing is the edges of the textures, which would be out of sight during normal gameplay. So if you could zoom out far enough (is there a mod for that?) You'd see this in game as well

**Comment by Gfiti** [comment]
Hey. I just recently did a run without the Clean Textures mod that had alot of water. Sadly I got the artifacts again. So it doesn't seem to be clean textures. It's less noticable on a quick glance but the same shapes can be found. 
Example: https://i.gyazo.com/0febaf91bc9f69e8dafb9cfaf2379e17.jpg

**Comment by TheVillageGuy** [comment]
> Hmmm, it's not immediately obvious what's going on. I'd have to see how the mod is performing rendering, it looks like the capture happened in the middle of a water rendering cycle. If that's the case, it'd be up to the mod authors to check if a map rendering is happening and not start a new water render if so.

As it happens without the clean textures mod that rules that out, I guess. So if I get it right, the water textures change while the render is being made?

**Comment by TheVillageGuy** [comment]
> Hey. I just recently did a run without the Clean Textures mod that had alot of water. Sadly I got the artifacts again. So it doesn't seem to be clean textures. It's less noticable on a quick glance but the same shapes can be found. Example: https://i.gyazo.com/0febaf91bc9f69e8dafb9cfaf2379e17.jpg

Could you upload a render of that on https://rimworld.gallery/m/debug with a modlist in the comments, please

**Comment by Gfiti** [comment]
The Renders were quite too big for the site, I uploaded them to my google drive, hope thats okay for you

https://drive.google.com/drive/folders/1T_J6KMDpJnLzGIIyxVXh2UTmMxrPl_jC

Render 1 - Water Base with Vanilla Textures
Modlist: https://rentry.co/89x6b

Render 2 - Water Base with Clean Textures
Modlist: https://rentry.co/shuio

Render 3 - River with Clean Textures (renders totally flat water, like there is no flow data at all)
Modlist: Harmony, DLCS, Clean Textures, Progress Renderer

Render 4 - River with Vanilla Texures (renders totally flat water, like there is no flow data at all)
Modlist: Modlist: Harmony, DLCS, Progress Renderer

**Comment by TheVillageGuy** [comment]
> > Hey. I just recently did a run without the Clean Textures mod that had alot of water. Sadly I got the artifacts again. So it doesn't seem to be clean textures. It's less noticable on a quick glance but the same shapes can be found. Example: https://i.gyazo.com/0febaf91bc9f69e8dafb9cfaf2379e17.jpg
> 
> Could you upload a render of that on https://rimworld.gallery/m/debug with a modlist in the comments, please


> The Renders were quite too big for the site, I uploaded them to my google drive, hope thats okay for you

I assume you tried uploading pngs, if you upload jpgs you should be fine

**Comment by Gfiti** [comment]
Here is an update, I just talked to TheVillageGuy on Discord for a bit.

First of all here is a full Render of one of my bases, vanilla textures, modlist: https://rentry.co/89x6b
https://rimworld.gallery/m/debug/t/514/Render-1-Water-Base-with-Vanilla-Textures
(It's the same picture as Render 1 I had on my drive)

I tested some things, here are the results:

- Removing zones (home zones, stockpiles etc) had no effect
- Removing roofs also had no effect
- Rendering as png or jpg made no difference (render settings: https://i.gyazo.com/67d343e1eafd2747444417d181a4277d.png )
- Artifacts are on different locations even if the same save is loaded and the base is rendered at the exact same day
- Resolution played on makes no difference (I usually play on 4k, 1080 showed the same results)
- **Creating a new map without mods (besides Progress Renderer) also made no difference, the artifacts can still be found.**

Assumption from TheVillageGuy:
"I think it may have something to do with the way the map is stored in memory, there are probably layers influencing the way water reflections work that somehow repeat themselves or are offset in a different way.
Maybe, this doesn't only happen on water, but just harder to distinguish on other terrains, or even invisible.
If we're lucky we can find that layer and can disable it during rendering, but that's up to neptimus

I don't think it's something that can easily be solved but maybe if we make some clear examples we can ask the mod community
First lists see what neptimus thinks, however"

**Comment by JonathanTroyer** [comment]
No longer an issue in 1.5

**Comment by TheVillageGuy** [comment]
It seems it still is: https://media.rimworld.gallery/dc/19/dc19b4655b26063343420a38a84537fc1eb6cbb144db4c1dc5d53d2a64d36f51.jpg

**Comment by JonathanTroyer** [comment]
Found in the wild: https://www.reddit.com/r/RimWorld/comments/1bj8xo9/meet_ferdok_a_prosperous_port_town/

**Comment by JonathanTroyer** [comment]
This issue is really hard to consistently reproduce. I'm going to have to dive into RimWorld's shader code to figure it out.

**Comment by JonathanTroyer** [comment]
Lol pause the game at a tick before rendering you midwit.
